File: youpinhui-content-interface/src/main/java/com/youpinhui/content/service/ContentCategoryService.java
package com.youpinhui.content.service;

import java.util.List;

import com.youpinhui.entity.PageResult;
import com.youpinhui.pojo.TbContentCategory;


/**
 * provider interface
 * @author Administrator
 *
 */
public interface ContentCategoryService {

	/**
	 * find all
	 * @return
	 */
	public List<TbContentCategory> findAll();
	
	
	/**
	 * query according pagination 
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * Add
	*/
	public void add(TbContentCategory contentCategory);
	
	
	/**
	 * update
	 */
	public void update(TbContentCategory contentCategory);
	

	/**
	 * get one by id
	 * @param id
	 * @return
	 */
	public TbContentCategory findOne(Long id);
	
	
	/**
	 * batch delete
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * fuzzy search 
	 * if the object  is not null , get the parameter from it as the search condition.
	 * @param pageNum current page number
	 * @param pageSize size of the page to display
	 * @return
	 */
	public PageResult findPage(TbContentCategory contentCategory, int pageNum,int pageSize);
	
}

File: youpinhui-manager-web/src/main/webapp/js/service/contentCategoryService.js
//Sevice
app.service('contentCategoryService',function($http){
	    	
	 //test case , get all the data return a Json
	this.findAll=function(){
		return $http.get('../contentCategory/findAll.do');		
	}
	//pagination
	this.findPage=function(page,rows){
		return $http.get('../contentCategory/findPage.do?page='+page+'&rows='+rows);
	}
	//find one object
	this.findOne=function(id){
		return $http.get('../contentCategory/findOne.do?id='+id);
	}
	//add 
	this.add=function(entity){
		return  $http.post('../contentCategory/add.do',entity );
	}
	//update 
	this.update=function(entity){
		return  $http.post('../contentCategory/update.do',entity );
	}
	//batch delete
	this.dele=function(ids){
		return $http.get('../contentCategory/delete.do?ids='+ids);
	}
	//search
	this.search=function(page,rows,searchEntity){
		return $http.post('../contentCategory/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});
